mage/query-modules/python/gnn-link-prediction.md

```diff
@@ -0,0 +1,375 @@
+
+#### Input
+- `src_vertex: mgp.Vertex` → Source node.
+- `dest_vertices: List[mgp.Vertex]` → destination nodes. If they are not of the same type, an exception is thrown.
+- `k: int` → Number of edges to recommend. 
+
+#### Output
+- `score: mgp.Number` → Score between 0 and 1 that represents the probability of two nodes being connected.
+- `recommendation: mgp.Vertex` → A reference to the target node.
+
+```
+MATCH (v1:Customer {id: "8779-QRDMV"})
+MATCH (p:Plan) 
+WITH collect(p) AS all_plans, v1 
+CALL link_prediction.recommend(v1, all_plans, 5)
+YIELD score, recommendation
+RETURN v1, score, recommendation;
+```
+
+### **Load context**
+
+Loading the context means loading the model and the predictor. If the user specifies the path, the method will try to load it from there. Otherwise, context will be loaded from the default parameter specified in the link_prediction_parameters module.
+
+#### Input
+- `path: str` → Path to the folder where the model and the predictor are saved.
+
+#### Output
+- `status: mgp.Any` → True to indicate that execution went well.
+
+```
+CALL link_prediction.load_context() YIELD * RETURN *;
+```
+
+### **Reset parameters**
+
+User can explicitly reset parameters whenever he/she wants. Note, however, that parameters will be reset before the training even if not specified by the user because of implementation reasons.
+
+#### Output
+- `status: mgp.Any` → True to indicate that method is successfully finished.
+
+```
+CALL link_prediction.reset_parameters() YIELD * RETURN *;
+```
+
+## Results
+
+We extensively tested our model on [**CORA**](https://paperswithcode.com/dataset/cora) dataset and the Telecom recommendation dataset. Here are the results of one of our basic models on the Cora dataset after training for 10 epochs:
+
+| epoch_num | AUC   | accuracy | precision | recall | f1    |
+| --------- | ----- | -------- | --------- | ------ | ----- |
+| 1         | 0.64  | 0.594    | 0.613     | 0.494  | 0.547 |
+| 2         | 0.781 | 0.696    | 0.711     | 0.663  | 0.686 |
+| 3         | 0.798 | 0.729    | 0.752     | 0.682  | 0.715 | 
+| 4         | 0.754 | 0.686    | 0.716     | 0.617  | 0.663 |
+| 5         | 0.789 | 0.711    | 0.715     | 0.7    | 0.707 |
+| 6         | 0.813 | 0.756    | 0.742     | 0.784  | 0.763 |
+| 7         | 0.884 | 0.772    | 0.764     | 0.791  | 0.775 |
+| 8         | 0.859 | 0.775    | 0.781     | 0.766  | 0.773 |
+| 9         | 0.871 | 0.805    | 0.822     | 0.777  | 0.798 |
+| 10        | 0.832 | 0.759    | 0.776     | 0.729  | 0.752
+
+## Example
```

Comment:
SOme intro for this example... what is it an example of and what does it do and why :D